prefect_census/runs.py

```diff
@@ -0,0 +1,165 @@
+"""Module containing tasks and flows for interacting with Census sync runs"""
+import asyncio
+from enum import Enum
+from typing import Tuple
+
+from httpx import HTTPStatusError
+from prefect import flow, task
+from prefect.logging import get_run_logger
+
+from prefect_census.credentials import CensusCredentials
+from prefect_census.utils import extract_user_message
+
+
+class CensusSyncRunFailed(RuntimeError):
+    """Raised when unable to retrieve dbt Cloud run"""
+
+    pass
+
+
+class CensusSyncRunTimeout(RuntimeError):
+    """
+    Raised when a triggered job run does not complete in the configured max
+    wait seconds
+    """
+
+    pass
+
+
+class CensusGetSyncRunInfoFailed(RuntimeError):
+    """Used to idicate retrieve sync run info."""
+
+    pass
+
+
+class CensusSyncRunCancelled(Exception):
+    """Raised when a triggered sync run is cancelled"""
+
+    pass
+
+
+class CensusSyncRunStatus(Enum):
+    """Census sync statuses."""
+
+    CANCELLED = "cancelled"
+    WORKING = "working"
+    FAILED = "failed"
+    COMPLETED = "completed"
+
+    @classmethod
+    def is_terminal_status_code(cls, status_code: str) -> bool:
+        """
+        Returns True if a status code is terminal for a job run.
+        Returns False otherwise.
+        """
+        return status_code in [
+            cls.CANCELLED.value,
+            cls.FAILED.value,
+            cls.COMPLETED.value,
+        ]
+
+
+@task(
+    name="Get Census sync run details",
+    description="Retrieves details of a Census sync run"
+    "for the sync with the given sync_id.",
+    retries=3,
+    retry_delay_seconds=10,
+)
+async def get_census_sync_run_info(credentials: CensusCredentials, run_id: int):
```

Comment:
Should this be `census_credentials`
```suggestion
async def get_census_sync_run_info(census_credentials: CensusCredentials, run_id: int):
```

Reply:
it seemed redundant to me so I have it as `credentials` throughout